src/main/functions/DIDResolution.ts

```diff
@@ -31,6 +31,63 @@ export function getResolver(opts: ResolveOpts): Resolvable {
   }
 }
 
+/**
+ * This method returns a resolver object in OP/RP
+ * If the user of this library, configures OP/RP to have a customResolver, we will use that
+ * If the user of this library configures OP/RP to use a custom resolver for any specific did method, we will use that
+ * and in the end for the rest of the did methods, configured either with calling `addDidMethod` upon building OP/RP
+ * (without any resolver configuration) or declaring in the subject_syntax_types_supported of the registration object
+ * we will use universal resolver from Sphereon's DID Universal Resolver library
+ * @param customResolver
+ * @param subjectSyntaxTypesSupported
+ * @param resolverMap
+ */
+export function getResolverUnion(
+  customResolver: Resolvable,
+  subjectSyntaxTypesSupported: string[] | string,
+  resolverMap: Map<string, Resolvable>
+): Resolvable {
+  if (customResolver) {
+    return customResolver;
+  }
+  const fallbackResolver: Resolvable = customResolver ? customResolver : new UniResolver();
+  const uniResolvers: {
+    [p: string]: (did: string, _parsed: ParsedDID, _didResolver: Resolver, _options: DIDResolutionOptions) => Promise<DIDResolutionResult>;
+  }[] = [];
+  const subjectTypes: string[] = [];
+  if (subjectSyntaxTypesSupported) {
+    typeof subjectSyntaxTypesSupported === 'string'
+      ? subjectTypes.push(subjectSyntaxTypesSupported)
+      : subjectTypes.push(...subjectSyntaxTypesSupported);
+  }
+  if (subjectTypes.indexOf(SubjectSyntaxTypesSupportedValues.DID.valueOf()) !== -1) {
+    return customResolver ? customResolver : new UniResolver();
+  }
+  const specificDidMethods = subjectTypes.filter((sst) => !!sst && sst.startsWith('did:'));
+  specificDidMethods.forEach((dm) => {
+    let uniResolver;
+    if (!resolverMap.has(dm)) {
+      uniResolver = getUniResolver(getMethodFromDid(dm));
```

Comment:
I still would expect customResolver here if defined. Now the user is saying. This is my custom resolver. The implementation is doing. Oh see, we have a did method without a resolver, let's use the uni resolver. That is almost certainly not what the user intended to happen.

Reply:
yeah this is exactly what is happenning. look at line 50

Reply:
Yeah, that is not entirely what I had in mind. But let's leave it for now. I wanted the customResolver to be a fallback, but still allow a DID method to use another resolver. Current implementation just uses customResolver for everything if it is set.

I will create another ticket during the refactor tickets, for this